src/editor.js

```diff
@@ -1717,13 +1725,64 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    /**
+     * Convert valid url text into links inside the given element.
+     *
+     * @param {HTMLElement} el
+     */
+    _convertUrlInElement(el) {
+        // We will not replace url inside already existing Link element.
+        if (el.tagName === 'A') {
+            return;
+        }
+
+        for (let child of el.childNodes) {
+            if (child.nodeType === Node.TEXT_NODE && child.length > 3) {
+                const childStr = child.nodeValue;
+                const matches = getUrlsInfosInString(childStr);
+                if (matches.length) {
+                    // We only to take care of the first match.
+                    // The method `_createLinkWithUrlInTextNode` will split the text node,
+                    // the other url matches will then be matched again in the nexts loops of el.childnodes.
+                    this._createLinkWithUrlInTextNode(matches[0], child);
+                }
+            }
+        }
+    }
+
+    /**
+     * Create a Link in the node text based on the given data
+     *
+     * @param {Object} data
+     * @param {Node} textNode
+     */
+    _createLinkWithUrlInTextNode(data, textNode) {
+        setCursor(textNode, data.index, textNode, data.index + data.length);
+        this.document.execCommand('createLink', false, data.url);
+        this.document.getSelection().collapseToEnd();
+    }
+
     /**
      * Prevent the pasting of HTML and paste text only instead.
      */
     _onPaste(ev) {
         ev.preventDefault();
         const pastedText = (ev.originalEvent || ev).clipboardData.getData('text/plain');
-        this.execCommand('insertText', pastedText);
+        const splitAroundUrl = pastedText.split(URL_REGEX);
+        for (let i = 0; i < splitAroundUrl.length; i++) {
+            // Even indexes will always be plain text, and odd indexes will always be URL.
+            if (i % 2) {
+                const url = /^https?:\/\//gi.test(splitAroundUrl[i])
+                    ? splitAroundUrl[i]
+                    : 'https://' + splitAroundUrl[i];
+                this.execCommand(
+                    'insertHTML',
+                    '<a href="' + url + '" >' + splitAroundUrl[i] + '</a>',
```

Comment:
I believe we were instructed to use `target="_blank"` by default in Odoo, maybe it should be the case here too.

Reply:
Didn't know that, I'll change it ;)

Reply:
After Discord discussion this will not be done here, maybe in a future dev with clearer specs.